.travis.yml

```diff
@@ -53,7 +53,7 @@ matrix:
       # all of the dependencies are supported on 3.8.
       env:
         - TEST_ASCII="true"
-        - TEST_OPT_DEPENDENCY="matchpy numpy scipy gmpy2 matplotlib theano llvmlite autowrap cython wurlitzer python-symengine=0.5.1 tensorflow numexpr ipython antlr-python-runtime>=4.7,<4.8 antlr>=4.7,<4.8 cloudpickle pyglet pycosat lfortran python-clang lxml"
+        - TEST_OPT_DEPENDENCY="matchpy numpy scipy gmpy2 matplotlib theano llvmlite autowrap cython wurlitzer python-symengine=0.5.1 tensorflow numexpr ipython antlr-python-runtime>=4.7,<4.8 antlr>=4.7,<4.8 cloudpickle pyglet pycosat lfortran python-clang lxml lark-parser"
```

Comment:
I thought the grammar file doesn't depend on lark at runtime. Or are we testing that the grammar file generates properly somewhere?

Reply:
Any number of approaches, not mutually exclusive:
- don't check in the parser, if you want to use it off `git+https`, install lark
- build the parser at build time, ship, and use by default
  - while lark is fast, reading _another_ file from disk, generating the parser, etc. isn't free, especially if it were to be done every time
- if `lark` is available at runtime, and the user were to _want_ to change the grammar, they could

Reply:
So I think this is worth discussion. Lark can generate a python file `lark_grammar.py` in this case. Do we want to check in the generated file to git (currently I am doing this). This could also be done in the packaging step by sympy when it distributes a the sdist and wheels etc.

Reply:
Ahh I see you brought up this exact issue https://github.com/sympy/sympy/pull/19825#issuecomment-662766127. Yeup either works with me.

Reply:
I think we should either check it in or generate it at package build time. Generating it on the fly when the user calls the parser would be neat, but would probably slow things down more than necessary. 

The main disadvantage of checking it in is that people might not realize that it is a generated file, and try to edit it directly. On the other hand, not checking it in adds some complexity to the build process and makes things slightly harder for people who want to use SymPy directly from git without installing (currently SymPy doesn't have any build steps). My preference is toward checking the file in.

Reply:
No file is written to disk unless you explicitly _ask_ for it, with a special CLI invocation. It's purely a performance (and dependency-avoidance) mechanism.